thorlcr/thorutil/thbuf.cpp

```diff
@@ -1567,6 +1576,20 @@ class CSharedWriteAheadDisk : public CSharedWriteAheadBase
         tempFileIO->setSize(0);
         tempFileOwner->noteSize(0);
     }
+    virtual unsigned __int64 getStatistic(StatisticKind kind) const override
+    {
+        switch(kind)
+        {
+            case StSizeSpillFile:
+                return tempSizeSpilled;
+            case StTimeSpillElapsed:
+                return cycle_to_nanosec(tempFileElapsedCycles);
+            case StNumSpills:
+                return 1;
+            default:
+                return 0;
```

Comment:
in hashdedup getstatistic is implemented by overriding StSizeDiskWrite with StSizeSpillFile, and defaulting for other io stats to use the underlying ifileio (hashdedup statsmapping also has diskWriteRemoteStatistics).
Perhaps both should be based on diskLocalStatistics.

This needs revisiting, we need to collect all temp fileIO stats. (e.g. inc. read stats), and we need to distinguish them from non-temp stats (map them to new stat types).